src/Core/Building.java

```diff
@@ -0,0 +1,99 @@
+package Core;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public abstract class Building {
+    protected Map<String, Room> rooms = new HashMap<>();
+    protected List<Suspect> suspects = new ArrayList<>();
+    protected DoctorWatson watson;
+    protected Room currentRoom;
```

Comment:
Why does the building have a current room variable? 

A lot of these variables are also stored in CaseFile and GameContext, maybe move to a single place